### PR TITLE
ESR60 fixes

### PR DESCRIPTION
**Pull request by abranson**
- ESR60 seems to be able to process the NAL lengths itself before passing the frame to gmp-droid, so detect that.
- Small bug and tidy up in the way the frame duration values were cached. The first frame's duration was always lost during codec creation.
- Don't send InputDataExhausted on decoded frames unless we don't have any out in the codec. One is already sent when a frame has been pushed to it, and multiple calls seem to make GMP skip frames.

### CODE REVIEW

File: gmp-droid.cpp [hunk 1]
```diff
@@ -301,7 +306,6 @@ class DroidVideoDecoder : public GMPVideoDecoder
 
     // Reset state
     m_draining = false;
-    m_dur.clear ();
```
Comment (abranson) [inline gmp-droid.cpp:304]: This caused the first frame's duration to be lost

File: gmp-droid.cpp [hunk 2]
```diff
@@ -188,20 +188,25 @@ class DroidVideoDecoder : public GMPVideoDecoder
...
-          return;
-        }
-        // Write NAL start code over the lenth
-        static const uint8_t code[] = { 0x00, 0x00, 0x00, 0x01 };
-        const uint8_t *start_code = code + (4 - start_code_len);
-        memcpy (inputFrame->Buffer () + offset, start_code, start_code_len);
-        offset += start_code_len + len;
 
-        LOG (DEBUG, "Parsed nal unit of size " << len);
+        if (len == 1) {
```
Comment (abranson) [inline gmp-droid.cpp:192]: ESR60 seems to have processed this already. OpenH264 sdoesn't allow for this, so I'm not sure why it doesn't need this. Perhaps later versions of Gecko stopped doing it again.

File: gmp-droid.cpp [hunk 3]
```diff
@@ -441,11 +445,15 @@ class DroidVideoDecoder : public GMPVideoDecoder
     // Send the new frame back to Gecko
     m_callback->Decoded (frame);
     LOG (DEBUG, "ProcessFrame: Returning frame ts: " << ts << " dur: " << dur);
-    if (!m_draining)
-      m_callback->InputDataExhausted ();
-    LOG (DEBUG, "Buffers still out " << m_dur.size ());
-    if (m_draining && m_dur.size () == 0) {     // TODO: we never get the buffers down to 0
-      m_callback->DrainComplete ();
+    if (m_dur.size () == 0) {
```
Comment (abranson) [inline gmp-droid.cpp:448]: Doing InputDataExhausted too many times makes ESR60 skip frames even if they are decoded.

**Review by mlehtima — APPROVED** [review]

File: gmp-droid.cpp (hunk 2 above)
Comment (abranson) [reply gmp-droid.cpp:192]: ESR60 has special handling for openh264 to supply it with the same AVCC format as before. It's changed to Annex B for everything else.

**Pull request merged**